File: java/v2/hello/src/main/java/xyz/hoper/client/GrpcClientMananer.java
package xyz.hoper.client;

import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;
import org.springframework.stereotype.Component;

import java.util.concurrent.TimeUnit;

@Component
public class GrpcClientMananer {

    public ManagedChannel getChannel(String host, int port){
        ManagedChannel channel = ManagedChannelBuilder.forAddress(host,port)
                .usePlaintext()
                .disableRetry()
                .idleTimeout(5, TimeUnit.SECONDS)
                .build();
        return channel;
    }
}

File: server/kotlin/v2/protobuf/src/main/kotlin/xyz/hoper/hello/client/GrpcClientMananer.kt
package xyz.hoper.hello.client

import io.grpc.ManagedChannel
import io.grpc.ManagedChannelBuilder
import org.springframework.stereotype.Component
import java.util.concurrent.TimeUnit

@Component
class GrpcClientMananer {
    fun getChannel(host: String?, port: Int): ManagedChannel {
        return ManagedChannelBuilder.forAddress(host, port)
                .usePlaintext()
                .disableRetry()
                .idleTimeout(5, TimeUnit.SECONDS)
                .build()
    }
}